Develop All the functions, reconstruct and check

steward379 opened the pull request:
http://54.253.20.174:3000/

* All requirements mentioned in the document shall have been met.
* Use SASS to define variables.
* Apply interval Observer API to accomplish the scrolling task.

* ADDITONAL: You can use keyboard `Enter` button to submit `search query` even if you type in Mandarin since composition issue were also considered.

steward379 committed:
Change the API to mine
Adjust the UI so it looks similar to what's on Figma

steward379 committed:
Finish All the functional Aspect
- Search Query function
- MRT Button Search
    -* Make an API params change to show MrtOnly Data
- AutoScrolling

steward379 committed:
Optimization
- Optimize the CSS code
- Optimized the JS code so it shall be functional

steward379 committed:
- Add font CDN
- Extract Main.js
- Use SCSS
- Add fake /favicon.ico

steward379 committed:
Use SASS to reconstruct the CSS file
- Read Figma styleguides to create variables.
- Look up the components to meet trivials and details.

steward379 committed:
Add unobserve and options (root, rootMargin, threshold)
to IntersectionObserver

steward379 committed:
- Fix the problem of 'no-data' rerendering
- Try to fix the arrow button turn to oval when use cell phones.

cwpeng reviewed (approved):
Good. Done.

It will be better to implement scrolling animation effect **smoothly**.

steward379 committed:
Adjust css
- Adjuct the arrow button to the right place.
- Enable arrow buttons to jump several options, scroll-x a little bit more.
- #ADDITIONAL: Create bounce effect when scrolling to the left/right end.

steward379 commented:
I renewed a version in the very last revise.

Adjust css
- Adjuct the arrow button to the right place.
- Enable arrow buttons to jump several options, scroll-x a little bit more.
- #ADDITIONAL: Create bounce effect when scrolling to the left/right end.
